mlflow/models/signature.py

```diff
@@ -393,6 +395,15 @@ def _infer_signature_from_type_hints(
     if type_hints.input is None:
         return None
 
+    _pyfunc_decorator_not_used = getattr(func, "_is_pyfunc", None) is not True
+    if _pyfunc_decorator_not_used:
+        # stacklevel is 3 because we have a decorator
+        warnings.warn(
+            "Decorate your callable with `@mlflow.pyfunc.utils.pyfunc` to enable auto "
```

Comment:
I use warning here because I want users to always use @pyfunc to make sure the behavior is consistent, and I feel like users would ignore INFO. This warning only shows up for callables because for PythonModel we already wrap the predict function

Reply:
Got it, yeah not strong opinion here (I usually ignore info as well hahaha)